tests/test_assets.py

```diff
@@ -761,12 +818,16 @@ def test_lookup_generic(self):
         """
         Ensure that lookup_generic works with various permutations of inputs.
         """
-        with build_lookup_generic_cases(self.asset_finder_type) as cases:
-            for finder, symbols, reference_date, expected in cases:
-                results, missing = finder.lookup_generic(symbols,
-                                                         reference_date)
-                self.assertEqual(results, expected)
-                self.assertEqual(missing, [])
+        cases = build_lookup_generic_cases()
+        # Make sure we clean up temp resources in the generator if we don't
+        # consume the whole thing because of a failure.
+        self.add_instance_callback(cases.close)
+        for finder, inputs, reference_date, country, expected in cases:
+            results, missing = finder.lookup_generic(
+                inputs, reference_date, country,
+            )
+            self.assertEqual(results, expected)
+            self.assertEqual(missing, [])
```

Comment:
should we check some cases that we expect to fail?

Reply:
There's a separate test case for `test_lookup_generic_handle_missing` that exercises those code paths.

Reply:
found that, thanks